### PR TITLE
Allow timeout to be specified over the outgoing http connection

### PR DESCRIPTION
**Pull request by HughePaul**
Specified with timeout as an option to the model constructor or timeout as an option to a http action (save, fetch, delete) but this may get filtered out by url().
Returns an ETIMEDOUT error through the callbacks on timeout.

### CODE REVIEW

File: lib/model.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ _.extend(Model.prototype, {
 
             reqConf.headers = _.extend({
                 'Content-Type': 'application/json',
-                'Content-Length': data.length
+                'Content-Length': Buffer.byteLength(data)
```
Comment (gavboulton) [inline lib/model.js]: Is this related to the timeout?

File: lib/model.js [hunk 2]
```diff
@@ -115,9 +122,24 @@ _.extend(Model.prototype, {
...
+            }, settings.timeout || this.options.timeout);
+        }
+
         request.on('error', function(e) {
-            _callback(e);
+            if (timeoutOccured && e.code === 'ECONNRESET') {
+                e.code = 'ETIMEDOUT';
+                e.message = 'Connection timed out';
+                e.status = 504;
+            }
```
Comment (gavboulton) [inline lib/model.js]: Could you add a test for this custom error.

File: lib/model.js (hunk 1 above)
Comment (HughePaul) [inline lib/model.js]: Ah No, this should be in a separate pull request

File: lib/model.js (hunk 2 above)
Comment (HughePaul) [inline lib/model.js]: Added into the timeout error test

**Comment by gavboulton** [comment]
This looks good. @JoeChapman @easternbloc @daniel-ac-martin what do you think?

**Comment by gavboulton** [comment]
Anyone? @JoeChapman @easternbloc @daniel-ac-martin @joefitter

**Comment by joefitter** [comment]
LGTM - sorry I didn't see the @ mention before!

**Comment by easternbloc** [comment]
LGTM
